realm/realm-library/src/main/java/io/realm/internal/SharedGroup.java

```diff
@@ -293,6 +297,29 @@ public int hashCode() {
         }
     }
 
+    /**
+     * Waits for change committed by {@link SharedGroup} in other Thread.
+     *
+     * @return {@code true} if successfully detects change, {@code false} no change has detected otherwise.
+     */
+    public boolean waitForChange() {
+        if (isStoppedWaitingByUser) {
+            isStoppedWaitingByUser = false;
+            return false;
+        }
+        boolean changeStatus = nativeWaitForChange(nativePtr);
```

Comment:
is there a use case, where Core returns from a `waitForChange` even without a commit from another thread (error, interrupt etc.) If it's the case then `false` have 2 UC (1. the user explicitly called `stopWaitForChange` or 2,. Core returned somehow without a commit)